### ISSUE TITLE
Translation

### ISSUE THREAD

**Issue by WebCodePoet** [antedebaas/DMARC-Reports#182]
I have seen, that you already have included some strings read for translation but not all.
Are there plans to make this project full translatable?

**Comment by antedebaas** [comment]
Yes. The idea is to add translations to it but I haven’t been consistent with the implementation of it so that needs to be fixed first.

Update: this is now a tracking issue
the following components need checking/updating in the code to have their strings encapsulated in the translator:
- [x] Controllers
- [ ] Commands
- [ ] EventListeners
- [ ] Forms
- [ ] Repository
- [ ] Security
- [ ] Templates

Other tasks:
- [x] Find external translation service for opensource projects

**Comment by antedebaas** [comment]
@WebCodePoet what language can you translate into?

cc @g

**Comment by antedebaas** [comment]
@WebCodePoet what language can you translate into?

cc @ceskyDJ

**Comment by ceskyDJ** [comment]
What about to use some translation system (e.g., [Tolgee](https://tolgee.io/)) just from the start? It'd give access for translating to wide range of translators (even non-coders). It brings in-context information, so translations have better quality. What do you think? 

Anyway, I'm here for Czech translation, when everything is ready for translating.

**Comment by WebCodePoet** [comment]
@antedebaas I can translate to German and Spanish.

**Comment by antedebaas** [comment]
@ceskyDJ , @WebCodePoet 
lokalise.com have granted an free opensource licence for translations so i set up the project there.
you can join with this link: https://app.lokalise.com/public/943233106735f8aa5bd752.59947412/

not everything is translatable yet but this will be extended in the future; but there are keys you can translate already

**Comment by ceskyDJ** [comment]
> @ceskyDJ , @WebCodePoet lokalise.com have granted an free opensource licence for translations so i set up the project there. you can join with this link: https://app.lokalise.com/public/943233106735f8aa5bd752.59947412/
> 
> not everything is translatable yet but this will be extended in the future; but there are keys you can translate already

Nice, thanks! I'll look at it when I have a little free time.

**Comment by WebCodePoet** [comment]
@antedebaas I have done the translations, but how do we choose a language inside Viesti?

I also figured out, that git is missing some version infos for

templates/domains/modals.html.twig
validators.es.xlf

maybe you will add them.

**Comment by antedebaas** [comment]
@WebCodePoet i saw you did quite a bit of work, thank you.
i haven't finished with adding translations to all the parts yet (limited free time here).
So there will be more to translate soon.
i plan to release a new version once everything is translated into at least one other language.
i also have to make modifications so that language selection is possible.

What do you mean with version infos?

**Comment by ceskyDJ** [comment]
I'm swamped by school projects these days. I'm going to do a Czech translation during the Christmas holiday. I'm sorry, but I don't have any free time for such big translation job sooner.

**Comment by antedebaas** [comment]
@ceskyDJ dont worry. Just spend time when you can and want. There are no deadlines anyway

**Comment by WebCodePoet** [comment]
@antedebaas if I run the update.sh the script says "you have unversioned commits, please add versioning" an then puts out this two:

templates/domains/modals.html.twig
validators.es.xlf

I am not very good with git, but I have not change any files inside the directory, so it looks like a problem for me.

**Comment by antedebaas** [comment]
@WebCodePoet Those are files you only have locally and are not part of the project.
You can remove those (and rerun update.sh to be safe)

**Comment by antedebaas** [comment]
These unversioned files are on your machine only.  You must have done something with them yourself since git complains about those files not being part of the repository

**Comment by ceskyDJ** [comment]
I started translating to Czech (fixing those default strings from an unknown author, resp.). There are tons of generic strings that are not very high quality (I mean in English). Do you use those? I think it's a little against UX, as users can't clearly understand them. Most of them are very technical. I keep translating, but sometimes I go quite far from the original string in English, as it's really bad. I don't want to translate it entirely differently, so I let not the bad ones as they are, but in the end, the result won't be as good as it could be (I mean, even in an English version). The bonus of a big number of strings for a pretty small app is that there won't be many translators who would find the time to translate it. Just saying, it's your app…

**Comment by antedebaas** [comment]
Well I never said I was a good UX designer.
But you’re welcome to make improvements to the English strings. It’s in the code:)

**Comment by ceskyDJ** [comment]
Ok, sounds good! Thinking of these generic strings, aren't they from your PHP framework?

**Comment by antedebaas** [comment]
Some are yes. We can change those in the English “translation” if we want. 
The rest are in the code under src/ and in the templates

**Comment by ceskyDJ** [comment]
Ok, thanks for the info. I don't know, when I'll be able to finish the Czech translation, and edit these English strings, but I'll try to find some time in the following days to do it progressively.
